File: rb/features.rb
# Punkapi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module PunkapiFeatures
  def self.make_feature(name)
    case name
    when "base"
      PunkapiBaseFeature.new
    when "ratelimit"
      PunkapiRatelimitFeature.new
    when "retry"
      PunkapiRetryFeature.new
    when "test"
      PunkapiTestFeature.new
    when "timeout"
      PunkapiTimeoutFeature.new
    else
      PunkapiBaseFeature.new
    end
  end
end

File: py/features.py
# Punkapi SDK feature factory

from feature.base_feature import PunkapiBaseFeature
from feature.test_feature import PunkapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PunkapiBaseFeature(),
        "test": lambda: PunkapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
